src/converters/notion/index.ts

```diff
@@ -0,0 +1,232 @@
+    const keys = Object.keys(parsedContent)
+    const fields = keys.map(key => {
+        const newField = {
+            uid: idgenerator(),
+            name: key,
+            createdAt: new Date().getTime(),
+            editedAt: new Date().getTime(),
+            type: 'field',
+            todoState: undefined, 
+        }
+
+        const fieldChildren = [{
+            uid: idgenerator(),
+            name: parsedContent[key],
+            createdAt: new Date().getTime(),
+            editedAt: new Date().getTime(),
+            type: 'node',
+            todoState: undefined,
+        }]
+
+        return {...newField, children: fieldChildren}
+    }) as TanaIntermediateNode[]
+
+    return fields
+}
+
+const createRootNode = (parsedContent: Record<string, string>): TanaIntermediateNode => {
+    const newNode = {
+        uid: idgenerator(),
+        name: parsedContent['Name'],
+        createdAt: new Date().getTime(),
+        editedAt: new Date().getTime(),
+        type: 'node',
+        todoState: undefined,
+    } as const
+
+    const children = createChildrenForRoot(parsedContent)
+    return {...newNode, children}
+}
+
+
+const createRootLevelNodes = (parsedContent: Record<string, string>[]) => {
+    return parsedContent.map(createRootNode)
+}
+
+const countChildDecendents = (node: TanaIntermediateNode): number => {
+    if(!node.children){
+        return 0
+    }
+
+    return 1 + node.children.map(countChildDecendents).reduce((acc, cur) => acc + cur)
+ 
+}
+
+const countTotalNodes = (nodes: TanaIntermediateNode[]): number => {
+    const count = nodes.length
+    const childrenCount = nodes.map(countChildDecendents).reduce((acc, curr) => (acc + curr), 0)
+
+    return count + childrenCount
+}
+
+const isField = (node: TanaIntermediateNode) => node.type === 'field'
+
+const countFields = (nodes: TanaIntermediateNode[]): number => {
+    const count = nodes.map((node) => {
+        let fieldCount = 0
+        if(isField(node)) {
+            fieldCount += 1
+        }
+        return fieldCount + (node.children ? countFields(node.children) : 0)
+    }).reduce((acc, curr) => (acc + curr), 0)
+
+    return count
+}
+
+const countBrokenLinks = (nodes: TanaIntermediateNode[]): number => {
+    return nodes.map((node) => {
+        let count = 0
+        if(isPageLink(node.name)) {
+            count += 1
+        }
+        return count + (node.children ? countBrokenLinks(node.children) : 0)
+    }).reduce((acc, curr) => (acc + curr), 0)
+}
+
+const countLeafNodes = (nodes: TanaIntermediateNode[]) => {
+    return nodes.map(countChildDecendents).reduce((acc, curr) => acc + curr)
+}
+
+const createSummary = (nodes: TanaIntermediateNode[]) => {
+    return {
+        leafNodes: countLeafNodes(nodes),
+        calendarNodes: 0,
+        fields: countFields(nodes),
+        brokenRefs: countBrokenLinks(nodes),
+        topLevelNodes: nodes.length,
+        totalNodes: countTotalNodes(nodes)
+    }
+}
```

Comment:
this could definitely be more efficient (one reduce function) but I took the straight forward approach.